src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs

```diff
@@ -57,6 +57,11 @@ internal CompletionSource(ITextView textView, IThreadingContext threadingContext
             // We would like to be sure that nobody changes buffers at the same time.
             AssertIsForeground();
 
+            if (trigger.Reason == AsyncCompletionData.CompletionTriggerReason.Insertion && trigger.Character == '\t')
+            {
+                return AsyncCompletionData.CompletionStartData.DoesNotParticipateInCompletion;
+            }
```

Comment:
this needs an explanatory comment.  For example, just reading this seems off.  I read it as "we were triggered because the user inserted something, and then the user hit tab, but we don't want completion".  I realize that's not correct, but it's confusing nontheless :)

Note: it's also unclear why this is a problem.  when is `\t` a trigger character in the first place?  Shouldn't ShouldTriggerCompletion (below) just return `false` for this case?

Reply:
Agreed, why was this ever a trigger character?

Reply:
@AmadeusW why Tab is a trigger character?

Reply:
even if the editor calls into us for this... shouldn't we be filtering things out below because \t isn't a triger char for *roslyn*.

Reply:
I don't think there was any design whether tab should or should not be a trigger character. Was it a trigger character ever for C#\VB? Is it a trigger character for any other language?

Reply:
Not that i know of.  I don't mind it potentially being a trigger.  I could see some language actually using that in some manner.